[CS2103T-F12-4] Billboard

weiijiie opened the pull request:
@Davidcwh
@waifonglee
@waynefong0401
@xinmingzh

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+*MSS*
+
+1.  User requests to list expenses
```
Do you think that the listing step is redundant? What if the user already knows the index of the expense he wants to tag?

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+
+[Discrete]
+=== UC02: Filter expenses by tag
```
Maybe you could filter by other fields also?

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+
+[Discrete]
+=== UC06: Finding a record with a specified keyword
```
Maybe could use precondition of the keyword being valid characters (eg. English)

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 expenses/income records without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
Maybe can refine this NFR by indicating how typing is faster than using mouse? Quantitatively?

michelleykw commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |Tag a record | Categorize and better manage my records
+
+|`* * *` |user |Specify a time stamp on expenses|Know when I spend my money
```
Are you considering having a feature of notifying the user?

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |Forgetful user | Add additional details through a description for each record | Better keep track of the details of an expense/income
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |Visually inclined user |Attach an image to each record |Conveniently record additional details of the expense instead of typing it all out
```
Is this really a must-have? Can the software still maintain its function without this feature?

jcjjjared commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |Specify a time stamp on expenses|Know when I spend my money
+
+|`* * *` |user | Sort and filter records by category or tag | Know how my spending/income is distributed
```
Maybe can consider adding a search function?

michelleykw commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,60 +295,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user | Archive past records | Better manage current expenses
+
+|`* * *` | user | Unarchive records |
```
Would you consider having unarchive records as a must have, or should it be classified as nice to have?

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -142,72 +145,79 @@ image::StorageClassDiagram.png[]
 The following sequence diagram shows how the undo operation works:
 
+.Interactions Inside the Logic Component for the `undo` Command
 image::UndoSequenceDiagram.png[]
```
undoModel arrow should be dotted.

tysg replied:
Ditto.

waynefong0401 replied:
Oops, thank you for pointing out that.

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+
+These actions are facilitated by the `ArchiveWrapper` and `Archive` classes:
+
+*  `Archive` extends from `ExpenseList` in order to encapsulate an archive name and a list of expenses together as an archive.
```
Can consider adding a class diagram

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+Firstly, the user has the application running and has a non empty list of current expenses. The user can enter the `list` command to bring up
+this list. +
+
+Next, the user executes the command `archive add 3 arc/archiveName` to archive an expense. +
```
Might want to break this up into steps

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+
+// end::finding[]
+
+=== Statistics
```
No diagrams for statistics

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+
+Billboard will store all the command entered even the command is never executed.
+
+This function is facilitated by the `CommandHistory` class.
```
No information about CommandHistory. Could add a class diagram

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+The CommandHistory will be initialized with no command state,
+and the statePointer is not pointing to any command state.
+
+.Initial state of CommandHistory
```
Seems like the diagram does not really serve a purpose

chowyiyin commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,401 @@ image::CommitActivityDiagram.png[]
+.Interactions Inside the Logic Component for the `delete` Command
+image::DeleteSequenceDiagram.png[]
+
+The LogicManager will store the command to CommandHistory before parsing it to ensure all commands are saved.
```
Could add design considerations

chowyiyin reviewed (commented):
Good job :D

heze8 commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+The archive feature supports the following actions:
+
+* Creating an archive
+* Added an expense to an archive
```
There's past tense here, but the rest of this list is present continuous.

heze8 commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+this list. +
+
+Next, the user executes the command `archive add 3 arc/archiveName` to archive an expense. +
+The command is first parsed by `BillboardParser` to determine what kind of general command it is. `archive` indicates it is an archive command so the remaining input is parsed through `ArchiveCommandParser`. +
```
This paragraph uses the same font and bolding for classes, arguments, commands, and quotations. This might be confusing to the reader as they might not know what each bolded word represents. Maybe separating out the font and standardizing would make it better.

michelleykw reviewed (commented):
For the figures under 3.2.1 in the Developer Guide, perhaps, it would be better to label the figures and split the full sequence diagram  into small parts as the words are too small to be seen without zooming in.

heze8 commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+`Storage` to serialize into JSON format and writes it into the JSON file.
+
+==== Design Considerations
+===== Aspect: Data Structure to hold archives in `ArchiveWrapper`
```
Neat and tidy, perhaps a table could also be considered.

tswuxia reviewed (commented):
- Section 2.6 has insufficient details
![image](https://user-images.githubusercontent.com/51690889/67545942-8183d380-f72d-11e9-8798-45f2b631d62a.png)

 - Section 2.4 2.5 the diagrams were not updated 
![image](https://user-images.githubusercontent.com/51690889/67545971-995b5780-f72d-11e9-9244-e26263d6e194.png)

 - It would be better to use different fonts for methods, arguments etc. in the description
![image](https://user-images.githubusercontent.com/51690889/67546014-b859e980-f72d-11e9-9a7b-bb48faf7d5aa.png)

heze8 reviewed (commented):
Overall, it's reasonably neat and clean, good luck on improving it.

Other comments are left in the specific section.

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,7 +100,7 @@ image::LogicClassDiagram.png[]
 *API* :
 link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
```
What's the difference between this diagram and the diagram under the architecture part? Seems like it's repeated..

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -142,72 +145,79 @@ image::StorageClassDiagram.png[]
 image::UndoRedoState4.png[]
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow ...
+Step 6. The user executes `clear`, which calls `VersionedBillboard#commit()`. Since the `currentStatePointer` is not pointing at the end of the `stateList`, all billboard states after the `statePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/buy ...` command. This is the behavior that most modern desktop applications follow.
```
If `clear` deletes all the states, then where does `current state` point to?

waynefong0401 replied:
The current state should point to bb3:Billboard, sorry for my typo. Just in case you are unclear about how it works, I will further explain it. If you already know how it works, please ignore below. If the user executes any other billboard editing command(add, delete etc), then all the state after the current state will be removed, then execute the command, and store it into the state list, current state pointer will point to that state.  And you will not able to redo cause the state pointer is pointing to the last state in the list.

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+`ArchiveWrapper` is used in `ModelManager` and its respective operations are called to access and manipulate archive expenses when an archive command is entered. +
+Such operations include:
+
+* `ArchiveWrapper#AddArchive(Archive)` - Adds the given archive to the current map of archive objects.
```
This information can be found in Java doc. We feel that it's a bit redundant to leave them here.

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+
+These interactions with `Model` by `AddArchiveCommand` can be shown in the cropped portion of the full sequence diagram below:
+
+image::AddArchiveCommandSequenceDiagram_executeCommand.png[]
```
Good diagram!!

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+
+Below is the full sequence diagram of the operation:
+
+image::AddArchiveCommandSequenceDiagram.png[]
```
The font is a bit too small

tysg commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
+
+The following sequence diagram shows how the adding tag operation works.
+
+image::AddTagSequenceDiagram.png[]
```
`retrieveTag`'s return object is not defined in this sequence diagram.

tysg replied:
also the lines should stop after the cross, though we are aware of the technical difficulty of doing so.

tysg reviewed (commented):
Good features presented, overall neat and detailed.

weiijiie committed:
Fixed javafx version and runtime incompatibility warning

weiijiie committed:
Changed CompletableFuture to javafx native Task

weiijiie committed:
Updated css for piechart slightly

Davidcwh committed:
Merge pull request #109 from AY1920S1-CS2103T-F12-4/generate-statistics-asynchronously

Generate statistics asynchronously

waynefong0401 committed:
added ppp for wayne

waynefong0401 committed:
Deleted dead code and change address book to billboard

weiijiie committed:
Added skeleton for new chart type, heatmap

waynefong0401 committed:
Delete Other project

xinmingzh committed:
Merge pull request #111 from AY1920S1-CS2103T-F12-4/delete-dead-code

WeiWen's PPP and Delete dead code

weiijiie committed:
Implemented generator for heatmap

weiijiie committed:
Implemented chart logic for heatmap

xinmingzh committed:
Init commit for recurring expenses

weiijiie committed:
Added proper axes and bubble points for the heat maps

weiijiie committed:
Cleaned code for HeatMapChart and fixed bug with starting month

weiijiie committed:
Update user guide with new ui and libraries used

weiijiie committed:
Fix checkstyle

weiijiie committed:
Updated user guide

weiijiie committed:
Merge branch 'master' into enhance-charts-further

Davidcwh committed:
Merge pull request #112 from AY1920S1-CS2103T-F12-4/enhance-charts-further

Enhance charts and updated user guide and README

weiijiie committed:
Updated README

weiijiie committed:
Merge remote-tracking branch 'origin/enhance-charts-further' into enhance-charts-further

Davidcwh committed:
Merge pull request #113 from AY1920S1-CS2103T-F12-4/enhance-charts-further

Update README image

weiijiie committed:
Fixed string converters output being off by one for month name

weiijiie committed:
Updated scale factor for heatmap data points

weiijiie committed:
Fixed off by one error for heatmap points and implemented tests

Tests implemented for MonthConverter and DayOfWeekConverter. Also,
implementation of those classes were updated to use a map for
conversion of string to value.

weiijiie committed:
Added error checks for ExpenseHeatMap

weiijiie committed:
Fix checkstyle

weiijiie committed:
Refactored HeatMap to be slightly simpler by removing redundant pair

xinmingzh committed:
Merge remote-tracking branch 'origin/master' into recurring-expenses

waynefong0401 committed:
debug up down key

waynefong0401 committed:
Merge branch 'master' into debug-up-down-key

waynefong0401 committed:
Merge pull request #115 from AY1920S1-CS2103T-F12-4/debug-up-down-key

debug up down key

waifonglee committed:
maintained tag structures in sampleutildata

waifonglee committed:
added error message for adding existing tags

waifonglee committed:
added error message for removing non existing tags

waifonglee committed:
edited message_tag_commands in tagcommandparser

waifonglee committed:
ameded checkstyle main errors

waifonglee committed:
amended some messages

Davidcwh committed:
Merge branch 'master' into david-PPP

Davidcwh committed:
Fixed undo/redo archive revert bug

Davidcwh committed:
User Guide: Removed Help [Command] section

Davidcwh committed:
User Guide: Updated archive section

Davidcwh committed:
Added stuff to introduction

waifonglee committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F12-4/main into ped-tagging

Davidcwh committed:
Fixed failing tests after bug fix

Davidcwh committed:
Merge pull request #185 from AY1920S1-CS2103T-F12-4/ped-tagging

Resolved some bugs caught during ped

Davidcwh committed:
Updated Summary of Contributions section

Davidcwh committed:
Merge branch 'master' into david-ppp

Davidcwh committed:
Basic content of Summary of Contributions section

waifonglee committed:
clean up tests

waifonglee committed:
edited some javadocs

Davidcwh committed:
Updated archive command message

waifonglee committed:
edited userguide

waifonglee committed:
edited addtag userguide

waifonglee committed:
edited addtag userguide

waifonglee committed:
finish tagging userguide

waifonglee committed:
edit tagging userguide

Davidcwh committed:
Added diagrams to archive section of User Guide

Davidcwh committed:
Implemented clone methods to get true deep copy of expenses - undo revert archive bug fix

Davidcwh committed:
Fix checkstyle

Davidcwh committed:
Minor edits

Davidcwh committed:
Removed unnecessary example

waifonglee committed:
ppp update

waifonglee committed:
updated ppp

Davidcwh committed:
Updated Developer Guide archive

Davidcwh committed:
Updated User Guide archive

Davidcwh committed:
Completed PPP draft

Davidcwh committed:
Updated User Guide for archive

Davidcwh committed:
Updated Developer Guide for archive

Davidcwh committed:
Finalised PPP

Davidcwh committed:
Changed code contributed link

Davidcwh committed:
Renamed ppp to davidcwh.adoc

waifonglee committed:
Merge pull request #186 from AY1920S1-CS2103T-F12-4/david-ppp

Bug fixes+ My PPP

waifonglee committed:
edited ppp, developer & userguide

waifonglee committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F12-4/main into ped-tagging

waifonglee committed:
amended link to code contributed in ppp

waifonglee committed:
deleted ppp pdf

Davidcwh committed:
Merge pull request #187 from AY1920S1-CS2103T-F12-4/ped-tagging

Clean up tests + ppp for waifong

xinmingzh committed:
Merge branch 'master' into recurring-expenses

# Conflicts:
#	docs/UserGuide.adoc

xinmingzh committed:
Implement list and add recurrence without undo/redo and save/load

xinmingzh committed:
Merge branch 'recurring-expenses' of https://github.com/AY1920S1-CS2103T-F12-4/main into recurring-expenses

xinmingzh committed:
Remove duplicate getClone method

weiijiie committed:
Wrote positive test cast for HeatMapGenerator

weiijiie committed:
Merge branch 'master' into enhance-charts-further

weiijiie committed:
Updated default sample expenses

weiijiie committed:
Implement grouping halfway

weiijiie committed:
Added functional interface for grouping function

weiijiie committed:
Updated archive displaying behavior and updated commands to follow

Expense display only switches between default view and archive view
when a command that affects either is made. Commands that do not
affect them will cause the displayed list to remain the same.

Commands that switch the view include:
* Add
* Edit
* Delete
* Clear
* Find
* List
* All archive commands besides ListArchiveNames
* All tag commands besides ListTag
* Undo
* Redo

Commands that do not switch the view include:
* Exit
* Help
* ListArchiveNames
* ListTag
* History
* DisplayStats

weiijiie committed:
Fixed exit/help command not updated properly

weiijiie committed:
Checkstyle fixes

weiijiie committed:
Codacy fix

weiijiie committed:
Fixed leaving out add archive command from the changes

weiijiie committed:
Updated all instances of "" in CommandResult constructors to constant

weiijiie committed:
Changed impl of VersionedBillboard to allow tracking of displayed list

weiijiie committed:
Undid erroneous edit of AddArchiveCommand

weiijiie committed:
Fixed AddArchiveCommand to have proper behavior for displaying expenses

waifonglee committed:
added authorship

waifonglee committed:
amended ppp symbol description

waifonglee committed:
Merge branch 'master' into ped-tagging

weiijiie committed:
Fixed undo and redo commands to copy the undone/redone model instead

Without copying, after undoing/redoing, updating the model will alter
the command history, leading to unpredictable results.

weiijiie committed:
[Chore] Coveralls bug

Davidcwh committed:
Merge pull request #194 from AY1920S1-CS2103T-F12-4/fix-undo-redo-bug

Fixed undo and redo commands to copy the undone/redone model instead

Davidcwh committed:
Merge branch 'master' into add-authorship

weiijiie committed:
Merge branch 'master' into change-archive-displaying-behavior

# Conflicts:
#	src/main/java/seedu/billboard/logic/commands/RedoCommand.java
#	src/main/java/seedu/billboard/logic/commands/UndoCommand.java

Davidcwh committed:
Merge branch 'master' into change-archive-displaying-behavior

Davidcwh committed:
Merge branch 'master' into ped-tagging

Davidcwh committed:
Merge pull request #192 from AY1920S1-CS2103T-F12-4/ped-tagging

Amended ppp

Davidcwh committed:
Merge branch 'master' into add-authorship

Davidcwh committed:
Amended ppp to match symbols and formatting in adoc and html

Davidcwh committed:
Added RepoSense authorship

weiijiie committed:
Merge branch 'master' into change-archive-displaying-behavior

weiijiie committed:
Merge remote-tracking branch 'origin/change-archive-displaying-behavior' into change-archive-displaying-behavior

# Conflicts:
#	src/main/java/seedu/billboard/logic/commands/RedoCommand.java
#	src/main/java/seedu/billboard/logic/commands/UndoCommand.java

weiijiie committed:
Fix bug

Davidcwh committed:
Merge pull request #191 from AY1920S1-CS2103T-F12-4/change-archive-displaying-behavior

Change archive displaying behavior

weiijiie committed:
Merge branch 'master' into enhance-charts-further

waynefong0401 committed:
Update some parts of the guides

waynefong0401 committed:
Merge branch 'master' into debug-undoredo

* master:
  Fix bug
  Fixed AddArchiveCommand to have proper behavior for displaying expenses
  Undid erroneous edit of AddArchiveCommand
  Changed impl of VersionedBillboard to allow tracking of displayed list
  Updated all instances of "" in CommandResult constructors to constant
  Fixed leaving out add archive command from the changes
  Codacy fix
  Checkstyle fixes
  Fixed exit/help command not updated properly
  Updated archive displaying behavior and updated commands to follow

Davidcwh committed:
Merge branch 'master' into add-authorship

Davidcwh committed:
amended ppp

waifonglee committed:
Merge pull request #195 from AY1920S1-CS2103T-F12-4/add-authorship

Added authorship + amended PPP

weiijiie committed:
Cleaned and fixed functionality in ExpenseTimelineChart

waynefong0401 committed:
Bugs solved

waynefong0401 committed:
Removed unused import

waynefong0401 committed:
Merge branch 'master' into debug-undoredo

Davidcwh committed:
Merge pull request #196 from AY1920S1-CS2103T-F12-4/debug-undoredo

Debug undoredo

Davidcwh committed:
Added fixed David's ppp html link in About Us file

Davidcwh committed:
Labelled Sorting in User Guide under v2.0

Davidcwh committed:
Updated Responsibilitiesin AboutUs.adoc

Davidcwh committed:
Fixed wei jie and xin ming ppp html links in AboutUs.adoc for when they add their ppp

Davidcwh committed:
Added screenshot for List commmand in UG

Davidcwh committed:
Edited List Command Screenshot

waynefong0401 committed:
Merge pull request #197 from AY1920S1-CS2103T-F12-4/documentation

Documentation changes

waynefong0401 committed:
Update UG

waynefong0401 committed:
Merge branch 'debug-undoredo' of https://github.com/AY1920S1-CS2103T-F12-4/main into debug-undoredo

* 'debug-undoredo' of https://github.com/AY1920S1-CS2103T-F12-4/main:
  amended ppp
  Added RepoSense authorship
  Amended ppp to match symbols and formatting in adoc and html

waynefong0401 committed:
Merge branch 'master' into debug-undoredo

* master:
  Edited List Command Screenshot
  Added screenshot for List commmand in UG
  Fixed wei jie and xin ming ppp html links in AboutUs.adoc for when they add their ppp
  Updated Responsibilitiesin AboutUs.adoc
  Labelled Sorting in User Guide under v2.0
  Added fixed David's ppp html link in About Us file

weiijiie committed:
Fixed chart issue where different series were not ordered properly

weiijiie committed:
Made logging and displayed message for changing charts more informative

weiijiie committed:
Fixed date formatting bug

weiijiie committed:
Refactored series management logic into its own class

weiijiie committed:
Extended groupby option functionality to heatmap

weiijiie committed:
Increased chart size slightly

waynefong0401 committed:
Update UG

waynefong0401 committed:
Restore CreateDateTime.java

weiijiie committed:
Added tests for FormattedDateConverter

weiijiie committed:
Updated HeatMapChart to support ExpenseGroupingOption

weiijiie committed:
Adjusted default options to work with all charts

weiijiie committed:
Added 2 new grouping methods, month and day-of-week

weiijiie committed:
Added tests for StatisticsFormatOptions

weiijiie committed:
Fixed import order

weiijiie committed:
Merge branch 'master' into enhance-charts-further

weiijiie committed:
Fixed bug in default expenses

weiijiie committed:
Updated xAxis tick values slightly

weiijiie committed:
Fixed large numbers glitching timeline chart display

weiijiie committed:
Added javadocs comments and fixed import order

weiijiie committed:
Deleted debugging message

Davidcwh committed:
Merge pull request #199 from AY1920S1-CS2103T-F12-4/debug-undoredo

Update UG

weiijiie committed:
Merge remote-tracking branch 'origin/master' into enhance-charts-further

Davidcwh committed:
Merge pull request #200 from AY1920S1-CS2103T-F12-4/enhance-charts-further

Enhance charts further

Davidcwh committed:
Merge branch 'master' into documentation

weiijiie committed:
Added validation for negative/zero values for amount

weiijiie committed:
Merge branch 'master' into improve-on-developer-guide-for-statistics

# Conflicts:
#	src/main/java/seedu/billboard/logic/commands/DisplayStatsCommand.java

weiijiie committed:
Updated user guide slightly

weiijiie committed:
Committed missing assets

weiijiie committed:
Wrote display stats user guide intro

weiijiie committed:
Added clear command section in user guide

weiijiie committed:
Edited incorrect line in intro and fixed add command formatting

weiijiie committed:
Finished timeline section for user guide

waifonglee committed:
change some formatting for userguide

waifonglee committed:
edited ppp format for userguide

waifonglee committed:
deleted ppp pdf

weiijiie committed:
Updated command format section

weiijiie committed:
Merge pull request #203 from AY1920S1-CS2103T-F12-4/ped-tagging

Edited Userguide

weiijiie committed:
Updated UG intro

weiijiie committed:
Merge branch 'master' into improve-on-developer-guide-for-statistics

# Conflicts:
#	docs/UserGuide.adoc

weiijiie committed:
Updated add command info

weiijiie committed:
Updated delete command in user guide

weiijiie committed:
Finished timeline stats portion

weiijiie committed:
Merge branch 'master' into update-input-validation-for-expenses

weiijiie committed:
Finished expense breakdown portion

Davidcwh committed:
Added instructions for manual testing for Archive

weiijiie committed:
Finished heatmap portion of user guide

waifonglee committed:
Merge pull request #201 from AY1920S1-CS2103T-F12-4/update-input-validation-for-expenses

Added validation for negative/zero values for amount

weiijiie committed:
Removed FAQ and added some lines about v2.0

Davidcwh committed:
Made formatting changes to instructions for manual testing for archive section

weiijiie committed:
Update line breaks slightly

weiijiie committed:
Fixed typo

Davidcwh committed:
Merge branch 'master' into documentation

weiijiie committed:
Added missing information in timeline section

weiijiie committed:
Merge branch 'master' into improve-on-developer-guide-for-statistics

waifonglee committed:
Merge pull request #202 from AY1920S1-CS2103T-F12-4/improve-on-developer-guide-for-statistics

Improve on user guide for statistics

waifonglee committed:
Merge branch 'master' into documentation

waifonglee committed:
Merge pull request #204 from AY1920S1-CS2103T-F12-4/documentation

Added Instructions of manual testing for Archive

waifonglee committed:
added manual testing instruction for tagging

waifonglee committed:
solve merge conflict

waifonglee committed:
edited some formatting problem in manual testing

waifonglee committed:
edited some formatting problem in manual testing for tagging

waifonglee committed:
edited manual testing for tagging in dev guide

weiijiie committed:
Edited introduction for developer guide

Davidcwh committed:
Fixed minor typo in ppp

waifonglee committed:
remove saving in instr for manual testing

Davidcwh committed:
Merge pull request #205 from AY1920S1-CS2103T-F12-4/ped-tagging

Instructions for manual testing for tagging

Davidcwh committed:
Minor change in formatting in User guide archive section

xinmingzh committed:
Add recurring expenses test cases

xinmingzh committed:
Merge branch 'master' into recurring-expenses

# Conflicts:
#	docs/UserGuide.adoc
#	src/main/java/seedu/billboard/model/ModelManager.java

xinmingzh committed:
Fix test

Davidcwh committed:
Merge branch 'master' into documentation

xinmingzh committed:
Fic checkstyletest

weiijiie committed:
Merge pull request #206 from AY1920S1-CS2103T-F12-4/documentation

Minor documentation changes

xinmingzh committed:
Fix codacy

xinmingzh committed:
Merge branch 'master' into recurring-expenses

xinmingzh committed:
Fix checkstyletest

weiijiie committed:
Updated overall flow for developer guide and added new chart

xinmingzh committed:
Fix bugs and javadoc typos

xinmingzh committed:
Fix codacy

xinmingzh committed:
Fix bugs mentioned in Wai Fong's review

weiijiie committed:
Completed developer guide

weiijiie committed:
Merge branch 'master' into improve-on-developer-guide-for-statistics

weiijiie committed:
Fixed some sentences

Davidcwh committed:
Merge pull request #207 from AY1920S1-CS2103T-F12-4/improve-on-developer-guide-for-statistics

Improve on developer guide for statistics

xinmingzh committed:
Merge branch 'master' into recurring-expenses

waynefong0401 committed:
Updated UGDG and PPP

waynefong0401 committed:
Merge branch 'master' into update-ug-dg-ppp-wayne

* master:
  Fixed some sentences
  Completed developer guide
  Updated overall flow for developer guide and added new chart
  Edited introduction for developer guide

waynefong0401 committed:
Merge branch 'master' into update-ug-dg-ppp-wayne

waynefong0401 committed:
Merge pull request #208 from AY1920S1-CS2103T-F12-4/update-ug-dg-ppp-wayne

Updated UGDG and PPP

weiijiie committed:
Fixed bug with empty expenses

xinmingzh committed:
Merge branch 'master' into recurring-expenses

xinmingzh committed:
Fix increment tag count

weiijiie committed:
Wrote major part of PPP

weiijiie committed:
Edited a left out area

waynefong0401 committed:
Merge branch 'master' into weijie-ppp

waynefong0401 committed:
Merge branch 'update-ug-dg-ppp-wayne' of https://github.com/AY1920S1-CS2103T-F12-4/main into update-ug-dg-ppp-wayne

* 'update-ug-dg-ppp-wayne' of https://github.com/AY1920S1-CS2103T-F12-4/main:

waynefong0401 committed:
Update PPP

Davidcwh committed:
Merge pull request #188 from AY1920S1-CS2103T-F12-4/recurring-expenses

Recurring expenses

waynefong0401 committed:
Merge branch 'master' into weijie-ppp

weiijiie committed:
Updated app version to 1.4

waynefong0401 committed:
Merge branch 'master' into update-ug-dg-ppp-wayne

* master:
  Fix increment tag count
  Fix bugs mentioned in Wai Fong's review
  Fix codacy
  Fix bugs and javadoc typos
  Fix checkstyletest
  Fix codacy
  Fic checkstyletest
  Fix test
  Add recurring expenses test cases
  Remove duplicate getClone method
  Implement list and add recurrence without undo/redo and save/load
  Init commit for recurring expenses

weiijiie committed:
Merge remote-tracking branch 'origin/weijie-ppp' into weijie-ppp

weiijiie committed:
Changed gif in ppp to png

waynefong0401 committed:
Merge pull request #210 from AY1920S1-CS2103T-F12-4/update-ug-dg-ppp-wayne

Update ug dg ppp wayne

weiijiie committed:
Merge branch 'master' into weijie-ppp

weiijiie committed:
Added missing file

weiijiie committed:
Merge remote-tracking branch 'origin/weijie-ppp' into weijie-ppp

Davidcwh committed:
Merge pull request #209 from AY1920S1-CS2103T-F12-4/weijie-ppp

PPP for weijie

weiijiie committed:
Edit erroneous information

waifonglee committed:
Merge branch 'master' into weijie-ppp

Davidcwh committed:
Merge pull request #211 from AY1920S1-CS2103T-F12-4/weijie-ppp

Weijie PPP Final edits

xinmingzh committed:
Update PPP

xinmingzh committed:
Merge remote-tracking branch 'origin/master' into ppp-xm

# Conflicts:
#	docs/DeveloperGuide.adoc

xinmingzh committed:
Fix

weiijiie committed:
Merge pull request #212 from AY1920S1-CS2103T-F12-4/ppp-xm

Xinming PPP UGDG
